source/Extensions/AmongUsExtensions.cs

```diff
@@ -39,5 +42,35 @@ public static class AmongUsExtensions
 
             return result;
         }
+
+        public static VisualAppearance GetDefaultAppearance(this PlayerControl player)
+        {
+            return new VisualAppearance()
+            {
+                ColorId = player.Data.ColorId,
+                HatId = player.Data.HatId,
+                SkinId = player.Data.SkinId,
+                PetId = player.Data.PetId
+            };
+        }
+
+        public static bool TryGetAppearance(this PlayerControl player, IVisualAlteration modifier, out VisualAppearance appearance)
+        {
+            if (modifier != null)
+                return modifier.TryGetModifiedAppearance(out appearance);
+
+            appearance = player.GetDefaultAppearance();
+            return false;
+        }
+
+        public static VisualAppearance GetAppearance(this PlayerControl player)
+        {
+            if (player.TryGetAppearance(Role.GetRole(player) as IVisualAlteration, out var appearance))
```

Comment:
```suggestion
            if (player.TryGetAppearance((IVisualAlteration) Role.GetRole(player), out var appearance))
```
don't really see `x as y` being used too much in ToU, although consistency isn't a big thing in this, just a small nit really

Reply:
Using the explicit cast changes the behavior. If the role is not an IVisualAlteration, the as operator will return null, allowing for the method to perform null checks.

Reply:
ah i see, didn't realise they behaved differently